applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py

```diff
@@ -219,7 +219,7 @@ def PrepareModelPart(self):
         # This will be removed once the Model is fully supported! => It wont e necessary anymore
         if not self.model.HasModelPart(self.main_model_part.Name):
```

Comment:
what I don't get is why you want to add the modelpart twice to the model
Here ...

Reply:
It is added just once, because the check is done "if not self.model.HasModelPart(self.main_model_part.Name):" before adding it

Reply:
Here is added in case that the problem uses restart